### PR TITLE
Fix Altoida mismatched headers

### PR DESCRIPTION
**Pull request by mpgxvii**
- Adds `FilePreProcessorFactory` to allow input stream pre-processing before the data is processed
- Adds `AltoidaSummaryCsvPreProcessor` to preprocess Altoida summary `export.csv` file, to allow incorrect headers to be fixed

### CODE REVIEW

**Comment by blootsvoets** [comment]
I would really prefer not to provide general extensive support for malformed CSV files. If we expect the file to be malformed, could we instead make a custom fileprocessor, not extending CsvLineProcessor? Alternatively, we could define file preprocessors to make file compatible with the CsvLineProcessor, but I'd like to extend that beyond just changing header names and values.

**Comment by mpgxvii** [comment]
> I would really prefer not to provide general extensive support for malformed CSV files. If we expect the file to be malformed, could we instead make a custom fileprocessor, not extending CsvLineProcessor? Alternatively, we could define file preprocessors to make file compatible with the CsvLineProcessor, but I'd like to extend that beyond just changing header names and values.

Okay sure I'll make the changes.

**Comment by mpgxvii** [comment]
> I would really prefer not to provide general extensive support for malformed CSV files. If we expect the file to be malformed, could we instead make a custom fileprocessor, not extending CsvLineProcessor? Alternatively, we could define file preprocessors to make file compatible with the CsvLineProcessor, but I'd like to extend that beyond just changing header names and values.

@blootsvoets For the custom fileprocessor, do you mean a processor for the whole file instead of a processor for just the header?

File: kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/altoida/summary/AltoidaSummaryCsvPreProcessor.kt [hunk 1]
```diff
@@ -0,0 +1,872 @@
...
+            return inputStream.bufferedReader().toCsvReader().use { reader ->
+                var header = reader.readNext()?.map { it }
+                        .takeIf { !it.isNullOrEmpty() }
+                        .orEmpty()
+                        .filter { h -> h.isNotEmpty() }
+
+                if (header.size < fileHeader.size) header = fileHeader
+                val line = reader.readNext()
+                val file = File.createTempFile("export", ".csv")
+                val outputStream = FileOutputStream(file)
```
Comment (blootsvoets) [inline kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/altoida/summary/AltoidaSummaryCsvPreProcessor.kt:54]: To ensure that we don't run into I/O issues or disk space filling up, could you make this a `ByteArrayOutputStream` instead and remove the file?

File: kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/altoida/summary/AltoidaSummaryCsvPreProcessor.kt [hunk 2]
```diff
@@ -0,0 +1,872 @@
...
+                val writer = CSVWriter(outputStream.bufferedWriter(StandardCharsets.UTF_8))
```
Comment (blootsvoets) [inline kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/altoida/summary/AltoidaSummaryCsvPreProcessor.kt:55]: For the `ByteArrayOutputStream`, no buffering is needed.

File: kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/altoida/summary/AltoidaSummaryCsvPreProcessor.kt [hunk 3]
```diff
@@ -0,0 +1,872 @@
...
+import org.radarbase.connect.upload.api.ContentsDTO
+import org.radarbase.connect.upload.api.RecordDTO
+import org.radarbase.connect.upload.converter.FileProcessorFactory
+import org.radarbase.connect.upload.converter.LogRepository
+import org.radarbase.connect.upload.converter.TopicData
+import org.slf4j.LoggerFactory
+import java.io.*
+import java.nio.charset.StandardCharsets
+
+class AltoidaSummaryCsvPreProcessor(
```
Comment (blootsvoets) [inline kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/converter/altoida/summary/AltoidaSummaryCsvPreProcessor.kt:32]: Please add a doc here about why this preprocessor is needed (changing the headers for empty fields?) and how it fixes it.

**Review by blootsvoets — APPROVED** [review]

**Pull request merged**